libs/contract/src/lib/avails/calendar/calendar.component.ts

```diff
@@ -161,18 +162,33 @@ export class AvailsCalendarComponent implements OnInit {
       const from = new Date(year + newState.start.row, newState.start.column);
       const to = new Date(year + newState.end.row, newState.end.column);
 
-      const parentMarker = this._availableMarkers.find(marker => {
-        // From the calendar pov range starts at the first day of the month
-        // but the avail term might not start at the first day of the month
-        const markerFromYear = marker.from.getFullYear();
-        const markerFromMonth = marker.from.getMonth();
-        const startDate = new Date(markerFromYear, markerFromMonth, 1).getTime();
-        return startDate <= from.getTime() && marker.to >= to;
-      });
-
-      if (!parentMarker) throw new Error(`Calendar Invalid Selection: a selection must be included in a marker!`);
-
-      this.selected.emit({ from, to, term: parentMarker.term, contract: parentMarker.contract });
+      /**
+       * If terms differ by media, then there is at least a territory
+       * that is common to all terms.
+       */
+      const termsDifferByMedia = this.avails.territories.some(territory => {
+        return this._availableMarkers.every(marker => marker.term.territories.includes(territory));
+      }); //otherwise terms differ by territory
+
+      const fields = termsDifferByMedia ? 'medias' : 'territories';
+      const cache: Record<string, { parentMarker: DurationMarker, list: string[] }> = {}
+      for (const field of this.avails[fields]) {
+        const marker = this._availableMarkers.find(({ term }) => {
+          return isCalendarTermInAvails(term, { ...this.avails, [fields]: [field] });
+        });
+        if (cache[marker.term.id]) cache[marker.term.id].list.push(field);
+        else {
+          cache[marker.term.id] = { list: [field], parentMarker: marker };
+        }
```

Comment:
Better do the opposite: 
```suggestion
        if (!cache[marker.term.id]) cache[marker.term.id] = { list: [], parentMarker: marker };
        cache[marker.term.id].list.push(field);
```